### DISCUSSION TITLE
Alternative starter to run DGS on spring-graphql core

### DISCUSSION THREAD

**Discussion by JamesPeters98** [Netflix/dgs-framework#1128] [Q&A]
Hi, I've been deciding whether to use DGS or the official Spring-GraphQL library and was watching this talk from Spring that mentions there will be an alternative starter provided for DGS to use the spring-graphql core. (https://youtu.be/Kq3UhUQdIO8?t=3160)

So my question is, what exactly is going to happen with the DGS framework. Will it eventually become on par with the official library in terms of features. Is it a bad idea to start a project using DGS in production if it is going to be made obsolete by the official library?

Will there be a need to migrate from DGS to the official library in the near future?

And what exactly will the alternative starter provide?

Ideally I would use DGS as it is more feature rich and seems to require less boiler plate code to get everything set up

**Comment by srinivasankavitha** [comment]
@JamesPeters98 - Thanks for checking out the project. We plan to make the DGS framework inter-operable with Spring-GraphQL in the future so users can ideally pick the features they want to use from each of the frameworks. We are not there yet and this will take some work which we plan to do later this year. We've done some initial prototyping but the effort requires more dedicated time to update it to current implementation.

We have been discussing with Spring GraphQL folks as well to coordinate our efforts. We anticipate that you should not need to migrate from DGS, rather be able to use what you want, e.g federation features, subscriptions, file uploads etc. 

All our internal services use the DGS framework and so do several others in the community, so it is entirely your preference at this point and what features you would like to leverage.
